control-plane/agents/src/bin/core/controller/io_engine/v1/nexus.rs

```diff
@@ -312,3 +312,19 @@ impl super::RpcClient {
         fetcher.get_nexus(node_id, nexus_id).await
     }
 }
+
+#[async_trait::async_trait]
+impl SnapshotApi for Nexus {
+    type CreateIn = CreateNexusSnapshot;
+    type CreateOutput = CreateNexusSnapshotResp;
+    async fn create_snapshot(
+        &self,
+        _request: &Self::CreateIn,
+    ) -> Result<Self::CreateOutput, SvcError> {
+        Err(SvcError::GrpcRequestError {
+            resource: ResourceKind::Replica,
+            request: "create_snapshot for nexus".to_string(),
```

Comment:
Resource kind already set above
```suggestion
            request: "create_snapshot".to_string(),
```